src/settings.cpp

```diff
@@ -378,27 +377,27 @@ std::string ManyOfManySettingDesc::FormatValue(const void *object) const
  * @param str Input string that will be parsed based on the type of desc.
  * @return The value from the parse string, or the default value of the setting.
  */
-size_t IntSettingDesc::ParseValue(const char *str) const
+size_t IntSettingDesc::ParseValue(std::string_view str) const
 {
-	char *end;
-	size_t val = std::strtoul(str, &end, 0);
-	if (end == str) {
+	StringConsumer consumer{str};
+	auto value = consumer.TryReadIntegerBase<size_t>(10);
```

Comment:
How are settings with negative values handled?
I suspect the old code relied on the ridiculous behavior of `strtoul`: https://godbolt.org/z/vbfYb7fY5
Some callers cast the result to `int32_t`.